### PR TITLE
Always create single subnet

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/roadmap/issues/1284

These changes require this PR https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/677

### CODE REVIEW

**Comment by fiunchinho** [comment]
The e2e tests fail because [my PR fixing a bug](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/664) is not included on the release that we are running.

File: helm/cluster-gcp/values.yaml [hunk 1]
```diff
@@ -16,8 +16,7 @@ gcp:
     - europe-west3-c
 
 network:
-  autoCreateSubnetworks: true
-  nodeSubnetCidrs: []  # Array of CIDRs to be used if `autoCreateSubnetworks` is set to `false`
+  nodeSubnetCidr: "192.168.0.0/24"  # CIDR to be used for the subnet
```
Comment (AverageMarcus) [inline helm/cluster-gcp/values.yaml:19]: Are we sure this is large enough for all cases?

**Review by AverageMarcus — APPROVED** [review]
LGTM, just a couple questions

File: helm/cluster-gcp/values.yaml (hunk 1 above)
Comment (fiunchinho) [reply helm/cluster-gcp/values.yaml:19]: I'll use `/16`

File: helm/cluster-gcp/templates/_gcp_cluster.tpl [hunk 2]
```diff
@@ -12,15 +12,13 @@ metadata:
...
     subnets:
-    {{- range .Values.network.nodeSubnetCidrs }}
-    - cidrBlock: {{ . }}
+    - name: {{ include "resource.default.name" $ }}-subnetwork
+      cidrBlock: {{ .Values.network.nodeSubnetCidr }}
       region: {{ $.Values.gcp.region }}
       routeTableId: false
-    {{- end }}
-    {{- end }}
+      enableFlowLogs: false
```
[the remark this answers is not in the document]
Comment (fiunchinho) [reply helm/cluster-gcp/templates/_gcp_cluster.tpl:21]: No customer has asked for it yet AFAIK.

**Pull request merged**